### PR TITLE
Add a flag to expose schema with region required

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

The flag allows to expose a new schema for KEB that makes the region parameter mandatory, but KEB will still accept requests without the region parameter.

Changes proposed in this pull request:

- Add a flag to expose schema with the region required parameter
- Add unit tests

**Related issue(s)**

See also https://github.com/kyma-project/control-plane/issues/3006

### CODE REVIEW

File: internal/broker/services_test.go [hunk 1]
```diff
@@ -115,6 +115,166 @@ func TestServices_Services(t *testing.T) {
 		assertPlansContainPropertyInSchemas(t, services[0], "oidc")
 		assertPlansContainPropertyInSchemas(t, services[0], "administrators")
 	})
+
+	t.Run("should containt the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is true and RegionParameterIsRequired is false", func(t *testing.T) {
```
Comment (MarekMichali) [inline internal/broker/services_test.go:119]: Any unit tests added in this PR may be removed after this feature goes live on all environments.

Comment (jaroslaw-pieszka) [inline internal/broker/services_test.go:119]:
```suggestion
	t.Run("should contain the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is true and RegionParameterIsRequired is false", func(t *testing.T) {
```

File: internal/broker/services_test.go [hunk 2]
```diff
@@ -115,6 +115,166 @@ func TestServices_Services(t *testing.T) {
...
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name", "region"})
+		}
+
+	})
+
+	t.Run("should containt the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is true and RegionParameterIsRequired is true", func(t *testing.T) {
```
Comment (jaroslaw-pieszka) [inline internal/broker/services_test.go:159]:
```suggestion
	t.Run("should contain the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is true and RegionParameterIsRequired is true", func(t *testing.T) {
```

File: internal/broker/services_test.go [hunk 3]
```diff
@@ -115,6 +115,166 @@ func TestServices_Services(t *testing.T) {
...
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name", "region"})
+		}
+
+	})
+
+	t.Run("should containt the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is false and RegionParameterIsRequired is true", func(t *testing.T) {
```
Comment (jaroslaw-pieszka) [inline internal/broker/services_test.go:199]:
```suggestion
	t.Run("should contain the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is false and RegionParameterIsRequired is true", func(t *testing.T) {
```

File: internal/broker/services_test.go [hunk 4]
```diff
@@ -115,6 +115,166 @@ func TestServices_Services(t *testing.T) {
...
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name", "region"})
+		}
+
+	})
+
+	t.Run("should containt the property 'required' with values [name] when ExposeSchemaWithRegionRequired is false and RegionParameterIsRequired is false", func(t *testing.T) {
```
Comment (jaroslaw-pieszka) [inline internal/broker/services_test.go:239]:
```suggestion
	t.Run("should contain the property 'required' with values [name] when ExposeSchemaWithRegionRequired is false and RegionParameterIsRequired is false", func(t *testing.T) {
```

File: internal/broker/services_test.go [hunk 5]
```diff
@@ -115,6 +115,166 @@ func TestServices_Services(t *testing.T) {
...
+		// then
+		require.NoError(t, err)
+		assert.Len(t, services, 1)
+		assert.Len(t, services[0].Plans, 5)
+
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name"})
```
Comment (jaroslaw-pieszka) [inline internal/broker/services_test.go:274]: Do you check if there is only `name` or if the list of required contains `name`?

Comment (MarekMichali) [reply internal/broker/services_test.go:274]: I check if there is only `name`

Comment (MarekMichali) [reply internal/broker/services_test.go:274]: For example, if we pass []string{"name","region"}, we will get that the schema doesn't contain the `region`. For example, if we pass an empty slice, we will get that the schema has more values than we passed.

File: internal/broker/services_test.go [hunk 6]
```diff
@@ -139,3 +299,25 @@ func assertPlanContainsPropertyInUpdateSchema(t *testing.T, plan domain.ServiceP
 		t.Errorf("plan %s does not contain %s property in Update schema", plan.Name, property)
 	}
 }
+
+func assertPlanContainsPropertyValuesInCreateSchema(t *testing.T, plan domain.ServicePlan, property string, wantedPropertyValues []string) {
```
Comment (jaroslaw-pieszka) [inline internal/broker/services_test.go:303]: Could we use some asserts from the library?
We just compare string with array of strings.

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**